sdks/python/setup.py

```diff
@@ -436,7 +436,9 @@ def get_portability_package_data():
           ],
           'gcp': [
               'cachetools>=3.1.0,<6',
-              'google-api-core>=2.0.0,<3',
+              # Temporary workaround until grpcio releases a fix for
+              # https://github.com/grpc/grpc/issues/36265
+              'google-api-core>=2.0.0,!=2.17.*,!=2.18.*,<3',
```

Comment:
Is this going to be fixed in google-api-core, or should we just pin to <2.17 until the fix is merged?

Reply:
it will be fixed in grpcio.

Reply:
a lower bound restriction is prone to cause dependency resolution down the road.

Reply:
a better fix would be to exclude the broken versions of grpcio, but that would revert a recent change to enable Py 3.12 support and I'd prefer to let people to experiment with Py 3.12 as early as possible to gather more feedback even before we officially support it.

Reply:
if grpc team produces a fix in 1.62.2, we can revert this change,  regenerate the dependencies and cherry-pick that for 2.56.0.

Reply:
My concern is mostly that if google-api-core releases 2.19 and people are on old grpcio this problem will still exist. I think you're probably right, that is the best option still though.

Hopefully the fix comes in soon and we can cherry-pick in a better patch